Queue.h

```diff
@@ -0,0 +1,33 @@
+#ifndef QUEU_H_INCLUDED
+#define QUEU_H_INCLUDED
+#include"Queue_Config.h"
+#include "STD_TYPES.h"
+
+#if QUEUE_TYPE == QUEUE_LINKED
+typedef struct node {
+    uint16_t key;
+    struct node* next;
+}node;
+
+typedef struct queue {
+    uint16_t size;
+    struct node *front, *rear;
+}queue_t;
+
+#elif QUEUE_TYPE == QUEUE_ARRAY
+typedef struct queuearray
+{
+uint16_t size;
+uint16_t arr[ARRAY_SIZE];
```

Comment:
make the type of array as a macro and make the user determine which type he want e.g. `typedef   QueueType uint16`